### DISCUSSION TITLE
we connect mqtt im Hintergrund laufen lassen

### DISCUSSION THREAD

**Discussion by Schnabel80** [tillsteinbach/WeConnect-mqtt#311] [Q&A] [answered]
Hallo zusammen,

ich habe alles nach der Anleitung auf einem Raspberry Pi4 installiert und es funktioniert auch soweit weit. Allerdings bekomme ich Immer nach einer gewissen Zeit keine neuen Daten mehr. (Ich glaube, wenn die Verbindung zum Pi via Terminal beendet wird)
Wie kann ich das Programm im Hintergrund dauerhaft ausführen?

Danke.

Gruß
Christian

**Comment by Domoberry** [comment]
With risk of providing the obvious answer: See [https://github.com/tillsteinbach/WeConnect-mqtt/tree/main/autostart/systemd](url) on how to use weconnect-mqtt as a service. Possibility to autostart on RPI4 (re)boot as well. This applies for RPI4 based systems.

**Reply by Schnabel80** [reply]
thank you! this was really difficult to finde for me, perhaps a link on the overview page could help.

**Comment by Schnabel80** [comment]
leider muss ich mich nochmal melden. nach befolgen der Anleitung oben, erhalte ich folgenden Fehler:

Jan 21 17:05:47 VWsFriends systemd[1]: Started weconnect-mqtt.service - WeConnect-mqtt.
Jan 21 17:05:47 VWsFriends python[915]: /usr/bin/python: No module named weconnect-mqtt
Jan 21 17:05:47 VWsFriends systemd[1]: weconnect-mqtt.service: Main process exited, code=exited, status=1/FAILURE
Jan 21 17:05:47 VWsFriends systemd[1]: weconnect-mqtt.service: Failed with result 'exit-code'.

**Comment by Domoberry** [comment, marked as answer]
I'm absolutely no expert on this, so these are just ideas based on the setup here as wel as similar challenges...
Could you please check the unit file /etc/systemd/system/weconnect-mqtt.service? Does it mention "... weconnect_mqtt ..." or "... weconnect-mqtt ..."? In my case it is "weconnect_mqtt" **and** it has the -m option to indicate the following argument should refer to a Python module rather than a Python file. After any change in the unit file, use "sudo systemctl daemon-reload".
Also check if $PATH contains "/home/pi/.local/bin:", this is added by the weconnect-mqtt installation, but requires an RPI restart.
@others: feel free to comment!

**Reply by Schnabel80** [reply]
your eyes a crazy good! "_" and everything is good. Thanks a lot...
